solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -431,12 +432,13 @@ public void register() {
       cache.setState(SolrCache.State.LIVE);
       infoRegistry.put(cache.name(), cache);
     }
-    metricManager = core.getCoreContainer().getMetricManager();
-    registryName = core.getCoreMetricManager().getRegistryName();
+    this.solrMetricsContext = core.getSolrMetricsContext().getChildContext(this);
```

Comment:
`SolrIndexSearcher` is not a `SolrMetricProducer` but it registers some gauges so I think it needs a child context - especially since its life-cycle is different from that of SolrCore.